Tasks/CopyPublishBuildArtifacts/copypublishbuildartifacts.ts

```diff
@@ -85,130 +86,139 @@ var artifactType: string = tl.getInput('ArtifactType');
+                    }
                 }
             }
         }
-    }
-    else {
-        tl.debug("Either contents or allFiles is empty");
-        files = allFiles;
-    }
+        else {
+            tl.debug("Either contents or allFiles is empty");
+            files = allFiles;
+        }
 
-    // copy the files to the staging folder
+        // copy the files to the staging folder
 
-    console.log("found " + files.length + " files");
-    if (files.length > 0) {
-        // make sure the staging folder exists
-        tl.mkdirP(stagingFolder);
+        console.log("found " + files.length + " files");
+        if (files.length > 0) {
+            // make sure the staging folder exists
+            tl.mkdirP(stagingFolder);
 
-        var commonRoot = getCommonLocalPath(files);
-        var useCommonRoot = !!commonRoot;
-        if (useCommonRoot) {
-            tl.debug("There is a common root (" + commonRoot + ") for the files. Using the remaining path elements in staging folder.");
-        }
+            var commonRoot = getCommonLocalPath(files);
+            var useCommonRoot = !!commonRoot;
+            if (useCommonRoot) {
+                tl.debug("There is a common root (" + commonRoot + ") for the files. Using the remaining path elements in staging folder.");
+            }
 
-        try {
-            var createdFolders = {};
-            files.forEach((file: string) => {
-                var stagingPath = stagingFolder;
-                if (useCommonRoot) {
-                    var relativePath = file.substring(commonRoot.length)
-                        .replace(/^\\/g, "")
-                        .replace(/^\//g, "");
-                    stagingPath = path.dirname(path.join(stagingFolder, relativePath));
+            try {
+                var createdFolders = {};
+                files.forEach((file: string) => {
+                    var stagingPath = stagingFolder;
+                    if (useCommonRoot) {
+                        var relativePath = file.substring(commonRoot.length)
+                            .replace(/^\\/g, "")
+                            .replace(/^\//g, "");
+                        stagingPath = path.dirname(path.join(stagingFolder, relativePath));
+                    }
+                    
+                    if (!createdFolders[stagingPath]) {
+                        tl.debug("Creating folder " + stagingPath);
+                        tl.mkdirP(stagingPath);
+                        createdFolders[stagingPath] = true;
+                    }
+                    
+                    tl.debug("Copying " + file + " to " + stagingPath);
+                    tl.cp("-Rf", file, stagingPath);
+                });
+
+                var data = {
+                    artifacttype: artifactType,
+                    artifactname: artifactName
+                };
+        
+                // upload
+                if (artifactType === "container") {
```

Comment:
The "else-if-filepath-artifact" block following this block was deleted. Now we short-circuit up front (before the copy) if it's a file path artifact.